### ISSUE TITLE
Attribute Grammar Thread

### ISSUE THREAD

**Issue by pengtianyue** [pengtianyue/Solvay_test#23]
Leave comments under this issue if there is additional attribute grammar needed.

**Comment by pengtianyue** [comment]
EMCMD: "Set 'WTR_CHG_EM' to [charge_command]". This will be parsed with compare operator "=". We can say: if "EM" in lhs and no 'path', return EMCMDAttribute(target_value = rhs).

**Comment by pengtianyue** [comment]
Calculate. It's usually included in Compare() like: " 'TIC-####' > ('OP005_VALUE' + 5) \* 1.05 - 10 ". Everything in the equation should be included in Calculate(), or there should be three Calculate() in the rhs of ">" that include lhs, op, and rhs. 

I prefer the first method: 1. use read() function of each element as an Attribute 2. replace the read value into the equation 3. use eval() to evaluate the entire equation.

**Comment by pengtianyue** [comment]
Elide Transition: #9

**Comment by pengtianyue** [comment]
Acquire. By saying "Acquire '[EM]' " in the diagram. We may have to think more on the Acquire step because failure on acquiring EMs won't cause a hold, but prompt OAR. it requires OAR response: release equipment and input OK, then it will loop back to acquire. I'm not sure if we should add this to the diagram because it's not specified in HLD and it would also fail the test. We can't really "force" an acquire failure to check if the prompt jumps out. Maybe we should put this into the Attribute types or test admin.

[Optional] Release. By saying "Release '[EM]' " in the diagram.

**Comment by mekcrab** [comment]
Agreed with "Acquire" being a viable new attribute type. I don't think anything "special" should be added to the test admin for corner cases, rather we need to handle them with attribute types. Then, we need only remap default OPC paths for attribute types if there are changes between projects - i.e. non-PCSD project.

**Comment by pengtianyue** [comment]
OAR response: need to set everything behind to force().

**Comment by pengtianyue** [comment]
OAR message: need string compare
